### PR TITLE
chore: Improve Java validation in openapi generator

### PR DESCRIPTION
**Pull request by marikaner**
Mini-PR to refactor the error handling in the java runtime validation.

### CODE REVIEW

File: packages/openapi-generator/src/cli.ts [hunk 1]
```diff
@@ -60,23 +60,25 @@ export = class GenerateOpenApiClient extends Command {
...
-    logger.error(
-      `Error in checking the java version. Is Java installed? This is mandatory for the OpenApi client generation. ${err.message}`
+    throw errorWithCause(
+      'Could not invoke `java` command. Probably no Java runtime is installed.',
+      err
+    );
+  }
+  if (response.exitCode !== 0 || !response.stderr.includes('version')) {
+    throw new Error(
+      `Could not validate Java version. Probably Java runtime is corrupt. Version check failed with "${response.stderr}"`
```
Comment (FrankEssenberger) [inline packages/openapi-generator/src/cli.ts:81]:
```suggestion
      `Could not validate Java version. Probably Java runtime is not installed or misconfigured. Version check failed with "${response.stderr}"`
```

Comment (FrankEssenberger) [reply packages/openapi-generator/src/cli.ts:81]: I think not installed or misconfigured like JAVA_HOME not properly set as env is the most common error.

**Review by FrankEssenberger — APPROVED** [review]
LGTM - consider my minor comment.

File: packages/openapi-generator/src/cli.ts [hunk 2]
```diff
@@ -60,23 +60,25 @@ export = class GenerateOpenApiClient extends Command {
...
-    logger.error(
-      `Error in checking the java version. Is Java installed? This is mandatory for the OpenApi client generation. ${err.message}`
+    throw errorWithCause(
+      'Could not invoke `java` command. Probably no Java runtime is installed.',
+      err
+    );
+  }
+  if (response.exitCode !== 0 || !response.stderr.includes('version')) {
+    throw new Error(
+      `Could not validate Java version. Probably Java runtime is not installed or misconfigured. Version check failed with "${response.stderr}"`
```
Comment (marikaner) [inline packages/openapi-generator/src/cli.ts:81]:
```suggestion
      `Could not validate Java version. Probably the Java runtime is not installed or misconfigured. Version check failed with "${response.stderr}"`
```

**Pull request merged**